src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1123,10 +1118,7 @@ impl LinkCollector<'_, '_> {
                     if fragment.is_some() {
                         anchor_failure(
                             self.cx,
-                            &item,
-                            path_str,
-                            dox,
-                            ori_link.range,
+                            diag_info,
```

Comment:
I think this might be a slight change in behaviour as `path_str != ori_link.link`. I don't remember the details but if this doesn't trip a test, maybe we should have one that covers it?

Reply:
It's a bug to ever pass in path_str here, otherwise it won't show the anchor at the end. Here's an example with the old diagnostic:

```
warning: `usize` contains an anchor, but links to builtin types are already anchored
 --> anchor.rs:1:6
  |
1 | //! [prim@usize#x]
  |      ^^^^^^^^^^^^ contains invalid anchor
  |
  = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default
```
I'll add a test.

FWIW, the reason I pass path_str separately from diag_info in some places is exactly so the caller isn't responsible for remembering to update it.